js/analyzescorer.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * inits the analyzerscorer used for scoring of the output from the textanalyzer
+ * @constructor
+ */
+YoastSEO_AnalyzeScorer = function( refObj ) {
+    this.__score = [];
+    this.refObj = refObj;
+    this.init();
+};
+
+/**
+ * loads the analyzerScoring from the config file.
+ */
+YoastSEO_AnalyzeScorer.prototype.init = function() {
+    this.scoring = analyzerScoring;
+};
+
+/**
+ * Starts the scoring by taking the resultObject from the analyzer. Then runs the scorequeue.
+ * @param resultObj
+ */
+YoastSEO_AnalyzeScorer.prototype.score = function( resultObj ) {
+    this.resultObj = resultObj;
+    this.runQueue();
+};
+
+/**
+ * runs the queue and saves the result in the __score-object.
+ */
+YoastSEO_AnalyzeScorer.prototype.runQueue = function() {
+    for ( var i = 0; i < this.resultObj.length; i++ ){
+        var subScore = this.genericScore( this.resultObj[i] );
+        if( typeof subScore !== "undefined" ) {
+            this.__score = this.__score.concat( subScore );
+        }
+    }
+    this.__totalScore = this.totalScore();
+};
+
+/**
+ * scoring function that returns results based on the resultobj from the analyzer matched with
+ * the scorearrays in the scoring config.
+ * @param obj
+ * @returns {{name: (analyzerScoring.scoreName), score: number, text: string}}
+ */
+YoastSEO_AnalyzeScorer.prototype.genericScore = function( obj ){
+    if( typeof obj !== "undefined" ) {
+        var scoreObj = this.scoreLookup( obj.test );
+        var score = { name: scoreObj.scoreName, score: 0, text: "" };
+        for ( var i = 0; i < scoreObj.scoreArray.length; i++ ) {
+            this.setMatcher( obj, scoreObj, i );
+            switch ( true ) {
+                case ( typeof scoreObj.scoreArray[i].type === "string" && this.result[scoreObj.scoreArray[i].type] ):
+                    return this.returnScore( score, scoreObj, i );
+                case ( typeof scoreObj.scoreArray[i].min === "undefined" && this.matcher <= scoreObj.scoreArray[i].max ):
+                    return this.returnScore( score, scoreObj, i );
+                case ( typeof scoreObj.scoreArray[i].max === "undefined" && this.matcher >= scoreObj.scoreArray[i].min ):
+                    return this.returnScore( score, scoreObj, i );
+                case ( this.matcher >= scoreObj.scoreArray[i].min && this.matcher <= scoreObj.scoreArray[i].max ):
+                    return this.returnScore( score, scoreObj, i );
+                default:
+                    break;
+            }
+        }
+        return score;
+    }
```

Comment:
What happens when object is undefined? In that case this function isn't returning anything, is that intended behavior?